osgi/src/main/java/cucumber/java/runtime/osgi/CucumberOSGi.java

```diff
@@ -0,0 +1,170 @@
+package cucumber.java.runtime.osgi;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.ParentRunner;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.event.TestRunFinished;
+import cucumber.api.java.ObjectFactory;
+import cucumber.api.junit.Cucumber;
+import cucumber.runtime.Backend;
+import cucumber.runtime.ClassFinder;
+import cucumber.runtime.Glue;
+import cucumber.runtime.Runtime;
+import cucumber.runtime.RuntimeOptions;
+import cucumber.runtime.RuntimeOptionsFactory;
+import cucumber.runtime.UnreportedStepExecutor;
+import cucumber.runtime.io.FileResourceLoader;
+import cucumber.runtime.io.MultiLoader;
+import cucumber.runtime.io.ResourceLoader;
+import cucumber.runtime.java.JavaBackend;
+import cucumber.runtime.junit.Assertions;
+import cucumber.runtime.junit.FeatureRunner;
+import cucumber.runtime.junit.JUnitOptions;
+import cucumber.runtime.junit.JUnitReporter;
+import cucumber.runtime.model.CucumberFeature;
+import cucumber.runtime.snippets.FunctionNameGenerator;
+import gherkin.pickles.PickleStep;
+
+/**
+ * <p>
+ * Classes annotated with {@code @RunWith(CucumberOSGi.class)} will run a Cucumber Feature
+ * inside an OSGi container using tycho.
+ * 
+ * @see Cucumber
+ * @see CucumberOptions
+ */
+public class CucumberOSGi extends ParentRunner<FeatureRunner> {

```

Comment:
Both `CucumberOSGi` and `Cucumber` implement the `ParentRunner`. We can't really deduplicate the setup in the constructor but  we can extract all the `children` related code to a parent class.